tsdb/engine/tsm1/data_file.go

```diff
@@ -177,7 +185,7 @@ func (e *IndexEntry) Contains(t time.Time) bool {
 
 func NewDirectIndex() TSMIndex {
 	return &directIndex{
-		blocks: map[string]indexEntries{},
+		blocks: map[string]*indexEntries{},
```

Comment:
`indexEntries` was `type indexEntries []*IndexEntry` before.  It's now a type and not a slice and since it's mutable I need to make it a pointer.